contracts/tribute/src/contract.rs

```diff
@@ -173,31 +173,34 @@ fn claim_tribute(
         )));
     }
 
-    // Look up voter's vote for the round, error if it cannot be found
-    let vote = query_user_vote(
+    let tribute = ID_TO_TRIBUTE_MAP.load(deps.storage, tribute_id)?;
+
+    // Look up voter's votes for the round, error if no votes can be found
+    let vote = match query_user_votes(
         &deps.as_ref(),
         &config.hydro_contract,
         round_id,
         tranche_id,
         voter.clone().to_string(),
-    )?;
+    )?
+    .into_iter()
+    .find(|vote| vote.prop_id == tribute.proposal_id)
+    {
+        None => {
+            // Error out if user didn't vote for the proposal that the given tribute belongs to.
+            return Err(ContractError::Std(StdError::generic_err(
+                "User voted for a different proposal than the one this tribute belongs to",
```

Comment:
```suggestion
                "User didn't vote for the proposal this tribute belongs to",
```
since a user can vote for multiple proposals now, "a different proposal" doesn't seem quite right